### ISSUE TITLE
"Cannot cast Newtonsoft.Json.Linq.JObject to Newtonsoft.Json.Linq.JToken" when starting docker compose debug with a function app project

### ISSUE THREAD

**Issue by ggirard07** [microsoft/DockerTools#421]
Hi all, not sure what is going on but I am getting this weird in a dialog form when I try to run my docker-compose solution with the debug enabled on an Azure function project in my solution. It used to work properly in the past. I updated Visual Studio and Docker Desktop a couple of times this the last time I touched this solution.
![image](https://github.com/microsoft/DockerTools/assets/2625886/64f4a707-fe42-44a8-b2ce-b525d6b66a72)

I am not able to reproduce on a fresh solution.
Running with the Azure function project as startup instead of docker compose, directly on the host, is running as expected.
Running with the Azure function project as startup instead of docker compose, within its own Docker container, has the same issue with a slightly different dialog.
![image](https://github.com/microsoft/DockerTools/assets/2625886/4d046222-b89d-430b-9537-9c7abc5c7edc)

I tried to update the build and run verbosity to diagnostic but I am not able to find any useful information (error is happening after the build is done `========== Build completed at X:XX PM and took XX.XXX seconds ==========`).
There is no useful information in Container Tools or Debug output panes.
The debug container is running in Docker Desktop but does not contain any log at that point yet.
I tried to get rid of all the obj/bin/.vs folders and also %LocalAppData%/vsdbg, %LocalAppData%/onecoremsvsmon and %LocalAppData%/\AzureFunctionsTools.

What should I do from that point to investigate the issue further?
As I am currently not able to tie up this error dialog to the docker tools or azure functions core tools or either Visual Studio itself...

Docker Desktop v4.27.2
Microsoft Visual Studio Professional v17.9.0

**Comment by NCarlsonMSFT** [comment]
@ggirard07 Based on code inspection I suspect we are failing to parse your local.settings.json file. Would you be able to provide a redacted version of the file so I can see its structure?

**Comment by ggirard07** [comment]
@NCarlsonMSFT this file and the `host.json` one where the 2 I initially suspected for this issue. I already tried to comment/remove most/all of their content but issue still persist. I also tried to replace their content with the content from the same files created by the fresh solution (the one working as expected I mentioned in my initial post) but issue also still persist.

I tried to disable most of the projects in the solution, except the function app. The container tools output is down to the following output when the error occurs:
```
docker exec -i fa4b865aa707 /bin/sh -c "if PID=$(pidof dotnet); then kill -9 $PID; fi; if PID1=$(pidof func); then kill -9 $PID1; fi"
========== Debugging ==========
docker ps --filter "status=running" --filter "label=com.docker.compose.service" --filter "name=^/My.Functions$" --format {{.ID}} -n 1
fa4b865aa707
```

**Comment by NCarlsonMSFT** [comment]
@ggirard07 I was able to repro this error dialog by adding a complex object in the values property which is not supported per [local-settings-file](https://learn.microsoft.com/en-us/azure/azure-functions/functions-develop-local#local-settings-file) can you verify that your local.settings.json only has string key value pairs for the values?

**Comment by ggirard07** [comment]
@NCarlsonMSFT I had just keys/values pair, except for the `Values` but just as the default template.
Here is the file content I am currently using, straight from the default template, which still reproduce the issue.
```
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "",
    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated"
  }
}

```

**Comment by ggirard07** [comment]
Dam it, I saw the light right after publishing my previous message: I have user secrets in this project! My user secrets indeed contain complex objects!

**Comment by NCarlsonMSFT** [comment]
Looks like you can flatten it: https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0&tabs=windows#json-structure-flattening-in-visual-studio

**Comment by NCarlsonMSFT** [comment]
Yes this is a regression from our adding support for fixing connections strings for the local emulator in user secrets.

**Comment by dbreshears** [comment]
This should be fixed in 17.10 Preview 3.
